### PR TITLE
ssm: Adding noDecrypt param support to SSM #10314

### PR DESCRIPTION
**Pull request by omerinvia**
Closes: [https://github.com/serverless/serverless/issues/10314](url)

### CODE REVIEW

File: lib/configuration/variables/sources/instance-dependent/get-ssm.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ module.exports = (serverlessInstance) => {
       });
       const region = !params || !params[0] || params[0] === 'raw' ? undefined : params[0];
       const shouldReturnRawValue = params && (params[0] === 'raw' || params[1] === 'raw');
+      const shouldSkipDecryption =
+        params && (params[0] === 'noDecrypt' || params[1] === 'noDecrypt');
```
Comment (medikoo) [inline lib/configuration/variables/sources/instance-dependent/get-ssm.js:25]: To secure proper resolution of all params, I'll probably go that route:

```javascript
const shouldReturnRawValue = params.includes('raw');
const shouldSkipDecryption = params.includes('noDecrypt');
_.pull(params, ['raw', 'noDecrypt']);
const region = params[0];
```

**Review by medikoo — CHANGES_REQUESTED** [review]
@omerinvia great thanks for that PR. It looks very good, I have just one suggestion towards param resolution

File: lib/configuration/variables/sources/instance-dependent/get-ssm.js (hunk 1 above)
Comment (omerinvia) [reply lib/configuration/variables/sources/instance-dependent/get-ssm.js]: wow, awesome suggestion

**Review by medikoo — COMMENTED** [review]
@omerinvia looks very good! I've just proposed as slight improvement to docs

File: docs/providers/aws/guide/variables.md [hunk 2]
```diff
@@ -340,6 +340,10 @@ New variable resolver, ensures that automatically other types as `SecureString`
 
 All `SecureString` type parameters are automatically decrypted, and automatically parsed if they export stringified JSON content (Note: you can turn off parsing by passing `raw` instruction into variable as: `${ssm(raw):/path/to/secureparam}`, if you need to also pass custom region, put it first as: `${ssm(eu-west-1, raw):/path/to/secureparam}`)
 
+#### Disable automatically `SecureString` decryption
+
+as mentioned above, by default , `SecureString` type parameters are automatically decrypted. in order to get the encrypted content , you can pass `noDecrypt` instruction into variable as: `${ssm(noDecrypt):/path/to/secureparam}`, if you need to also pass custom region, put it first as: `${ssm(eu-west-1, noDecrypt):/path/to/secureparam}`
+
```
[the remark this answers is not in the document]
Comment (omerinvia) [reply docs/providers/aws/guide/variables.md]: cool :) done

File: docs/providers/aws/guide/variables.md [hunk 3]
```diff
@@ -340,6 +340,12 @@ New variable resolver, ensures that automatically other types as `SecureString`
 
 All `SecureString` type parameters are automatically decrypted, and automatically parsed if they export stringified JSON content (Note: you can turn off parsing by passing `raw` instruction into variable as: `${ssm(raw):/path/to/secureparam}`, if you need to also pass custom region, put it first as: `${ssm(eu-west-1, raw):/path/to/secureparam}`)
 
+#### Disable automatically `SecureString` decryption
+
+as mentioned above, by default , `SecureString` type parameters are automatically decrypted.
```
Comment (medikoo) [inline docs/providers/aws/guide/variables.md:345]: Let's remove this lines

File: docs/providers/aws/guide/variables.md [hunk 4]
```diff
@@ -340,6 +340,10 @@ New variable resolver, ensures that automatically other types as `SecureString`
 
 All `SecureString` type parameters are automatically decrypted, and automatically parsed if they export stringified JSON content (Note: you can turn off parsing by passing `raw` instruction into variable as: `${ssm(raw):/path/to/secureparam}`, if you need to also pass custom region, put it first as: `${ssm(eu-west-1, raw):/path/to/secureparam}`)
 
+#### Disable automatically `SecureString` decryption
```
Comment (medikoo) [inline docs/providers/aws/guide/variables.md:343]: Let's remove this title

**Comment by omerinvia** [comment]
can you please go over the docs and lets try to have one comment for the docs :D (btw it was git copilot suggestion to write those dox)

**Review by medikoo — APPROVED** [review]
Thank you @omerinvia !

**Pull request merged**